platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -298,7 +303,105 @@ namespace platoon_strategic_ihp
         return msg;
     }
 
-    // compose platoon info msg for all states
+    // ----------------------------- UCLA: helper functions for cut-in from front -------------------------------//
+
+    // Generate a maneuver ID for new maneuver plans (only used for compose lane change maneuver)
+    std::string PlatoonStrategicIHPPlugin::getNewManeuverId() const {
+        /**
+         * @brief Function to generate new maneuver ID. (Use for genreating new lane change maneuver)
+         * 
+         * @return A newly generated uuid as int.
+         */
+
+        static auto gen = boost::uuids::random_generator(); // Initialize uuid generator
```

Comment:
Normally a static random number generator is initialized exactly once during a program's lifetime in order to maximize randomness.  Code as is will re-create the "gen" object and initialize on every call.  gen should be made a member variable and this statement moved to the constructor.